### PR TITLE
MAKE-862: implement wire protocol service

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/MAKE-862

* Define request/response types in `wire/io.go`.
* Implement wire protocol service with manager and process RPC handlers.

Still need to implement this with support for OP_MSG in case the shell is version 4.2 or higher.

### CODE REVIEW

File: wire/io.go [hunk 1]
```diff
@@ -0,0 +1,846 @@
...
+func intOK(ok bool) int {
+	if ok {
+		return 1
+	}
+	return 0
+}
+
+// ErrorResponse represents a response indicating whether the operation was okay
+// and errors, if any.
+type ErrorResponse struct {
```
Comment (tychoish) [inline wire/io.go]: seems like most of the reponse building code should end up in mrpc)

Comment (Kimchelly) [reply wire/io.go]: I don't think this makes sense given that all of the payloads are Jasper-specific and not defined in the mongo wire protocol spec.

**Comment by Kimchelly** [comment]
#134 is old, do not use.

**Comment by Kimchelly** [comment]
I updated this to use the shell service from mrpc.

File: wire/service.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	"github.com/evergreen-ci/mrpc"
+	"github.com/evergreen-ci/mrpc/mongowire"
+	"github.com/evergreen-ci/mrpc/shell"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/recovery"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+)
+
+// TODO: support jasper.RemoteClient functionality
```
Comment (tychoish) [inline wire/service.go]: is there a ticket for this?

File: wire/service.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+type service struct {
```
Comment (tychoish) [inline wire/service.go]: are we worried about thread safety or would a synchronized manager be enough?

File: wire/service.go (hunk 2 above)
Comment (Kimchelly) [reply wire/service.go]: https://jira.mongodb.org/browse/MAKE-1000

File: wire/service.go (hunk 3 above)
Comment (Kimchelly) [reply wire/service.go]: The thread safety guarantees for this is the same as for the REST/RPC managers. I'm not particularly concerned about having multiple connections given that the synchronized manager already locks all the manager methods.

**Pull request merged**